lib/Scvmm/miq_scvmm_vm_ssa_info.rb

```diff
@@ -0,0 +1,67 @@
+
+require 'miq_winrm'
+require 'miq_scvmm_parse_powershell'
+
+class MiqScvmmVmSSAInfo
+  attr_reader :vhd, :hostname, :checkpoints, :connection, :vhd_type
+  def initialize(provider, user, pass, port = nil)
+    @checkpoints = []
+    @vhd         = nil
+    @hostname    = nil
+    @vhd_type    = nil
+    @winrm       = MiqWinRM.new
+    winrmport    = port.nil? ? 5985 : port
+    options      = {:port     => winrmport,
+                    :user     => user,
+                    :pass     => pass,
+                    :hostname => provider
+                   }
+
+    @connection  = @winrm.connect(options)
+  end
+
+  def vm_host(vm_name)
+    host_script = <<-host_eol
```

Comment:
I thought you meant for the name of the ruby variable, making it a constant, which would be a 'no' since we need ruby variable substitution to happen each time they are run, but  you probably mean for the name of the heredoc start/end string, which would be a 'yes'.

With regard to the leading whitespace - that certainly makes sense and I will left align the heredocs.